### PR TITLE
Fix azure repo stream exhaust check for multipart uploads

### PR DESCRIPTION
**Pull request by albertzaharovits**
This PR fixes the validation of the conversion from an input stream to a flux in the
AzureBlobStore's multipart update logic, which erroneously checked that the upload
input stream is exhausted after each part's flux is completed. It should instead check
that the stream is only exhausted after the lat part's flux is completed.

I couldn't spot any unit tests for the AzureBlobStore's upload logic, so I've not added any
tests for this PR either.

### CODE REVIEW

File: plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java [hunk 1]
```diff
@@ -438,8 +449,21 @@ private void executeMultipartUpload(String blobName, InputStream inputStream, lo
             final List<String> blockIds = new ArrayList<>(nbParts);
             for (int i = 0; i < nbParts; i++) {
                 final long length = i < nbParts - 1 ? partSize : lastPartSize;
-                final Flux<ByteBuffer> byteBufferFlux =
-                    convertStreamToByteBuffer(inputStream, length, DEFAULT_UPLOAD_BUFFERS_SIZE);
+                Flux<ByteBuffer> byteBufferFlux = convertStreamToByteBuffer(inputStream, length, DEFAULT_UPLOAD_BUFFERS_SIZE);
+                if (i == nbParts - 1) {
+                    byteBufferFlux.doOnComplete(() -> {
+                        try {
+                            if (inputStream.available() > 0) {
```
Comment (original-brownbear) [inline plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java:456]: I wonder if we should just drop this check altogether instead? It seems kind of weird to check that we drained the stream fully when we also give the size we want to write (and the check relies on the fact that the stream reports `available()` as well which is not guaranteed).

Maybe we should just drop this check and only check that `currentTotalLength == blobSize` at the end? I think I'd prefer that option and it simplifies things?

**Review by original-brownbear — COMMENTED** [review]
Thanks for spotting this Albert. One question

File: plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java (hunk 1 above)
Comment (albertzaharovits) [reply plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java]: I'll take your suggestion to remove the check, although I'm torn between the complexity it adds in the code vs the aid in troubleshooting, but maybe the existing check about `currentTotalLength > blobSize` is sufficient.

**Review by original-brownbear — APPROVED** [review]
LGTM thanks Albert!

**Review by fcofdez — APPROVED** [review]
LGTM, thanks for fixing this Albert!

**Comment by albertzaharovits** [comment]
@elasticmachine update branch

**Pull request merged**